src/main/java/org/testng/internal/Parameters.java

```diff
@@ -43,6 +38,42 @@
  */
 public class Parameters {
   public static final String NULL_VALUE= "null";
+  private static Class<?>[] annotationList = new Class<?>[] {
+      BeforeSuite.class,
+      AfterSuite.class,
+      BeforeTest.class,
+      AfterTest.class,
+      BeforeClass.class,
+      AfterClass.class,
+      BeforeGroups.class,
+      AfterGroups.class,
+      BeforeMethod.class,
+      AfterMethod.class
+  };
+
+  private static Map<String, List<Class>> mapping = new HashMap<>();
```

Comment:
`private static Map<String, List<Class<?>>> mapping = Maps.new...();`
or better, if it works:
`private static Map<String, List<Class<? extends Annotation>>> mapping = Maps.new...();`